docs/user-guide/rug/event-handlers.md

```diff
@@ -0,0 +1,204 @@
+resulting from the expression execution.
+
+### Implementation
+
+Rug event handlers must implemented the `#!typescript HandleEvent<R,M>`, where
+the `R` and `M` type parameters refer to the expected root node and match types
+resulting from the execution of a [Path Expression][pes] respectively.
+
+### Discovery
+
+{!decorators/tags.md!}
+
+### EventPlans
+
+An `#!typescript EventPlan` describes the actions to be taken by the Rug runtime
+on behalf of the handler. EventPlans are composed of [Messages](#messages) and/or 
+[respondables](#respondables). Respondables instruct the rug runtime to immediately
+perform ordinary rug operations, whereas messages are sent to the Atomist Bot for
+display to the user.
+
+### Messages
+
+A `Message` represents presentable content and/or deferable actions displayed
+to the user in response to returning an `#!typescript EventPlan` from an event
+handler. Every message will end up being sent to a chat channel or user by the
+Atomist bot. However, each of the two available message types achieve this in 
+different ways.
+
+#### Lifecycle Messages
+
+{!lifecycle-message.md!}
+
+#### Directed Messages
+
+{!directed-message.md!}
+
+### Respondables
+
+An `#!typescript EventRespondable` is really just a container for an instruction and
+some optional `onError` and `onSuccess` capabilities. The `onError` and `onSuccess`
+properties of an `#!typescript EventRespondable` can be [messages](#messages), 
+[EventPlans](#eventplans) or [response handlers](#response-handlers).
+
+```typescript
+const plan = new EventPlan();
+plan.add(
+    {
+        instruction: {
+            kind: "edit",
+            project: "rugs",
+            name: "UpdateReadme",
+            parameters: {
+                newContent: "Rugs really tie the room together"
+            }
+        },
+        onSuccess: new DirectedMessage("woot", new ChannelAddress("#random")),
+        onError: {
+            kind: "respond",
+            name: "HandleReadmeUpdateErrors"
+        }
+    }
+)
+
+```
+
+The example above shows how to send `woot` to the `#random` channel using a
+[Directed Message](#messages) if the `edit` Instruction completes successfully,
+and invoke the `HandleReadmeUpdateErros` [Response Handler](#response-handlers)
+if it fails.
+
+### Instructions
+
+Instructions in an `#!typescript EventRespondable` have the following properties:
+
+*   `#!typescript kind: "generate" | "edit" | "execute"`: the kind of instruction 
+*   `#!typescript name: string`: the name of the operation to apply
+*   `#!typescript parameters: {}`: key/value pairs passed to the operation
+*   `#!typescript project?: string`: Project name (only for Generators & Editors)
+
+Instructions can be used to have the rug runtime run Rugs, such as invoking
+a Generator (`#!typescript "generate"`), an Editor (`#!typescript "edit"`), or
+a [Rug Funtion](#rug-functions) (`#!typescript "execute"`).
+
+### Rug Functions
+
+{!rug-functions.md!}
+
+!!! attention "Secrets"
```

Comment:
Perhaps this belongs in docs/common/secrets.md? If so, having the heading plus admonition title seems redundant.